lib/std/core/vector.kk

```diff
@@ -121,11 +184,44 @@ pub fun foreach-while( v : vector<a>, f : a -> e maybe<b> ) : e maybe<b>
     f(v.unsafe-idx(i))
 
 // Apply a function `f` to each element in a vector `v`
-pub fun map( v : vector<a>, f : a -> e b ) : e vector<b>
-  val w = unsafe-vector(v.length.ssize_t)
-  v.foreach-indexedz fn(i,x)
-    unsafe-assign(w,i,f(x))
-  w
+pub fip fun map( v : vector<a>, ^f : a -> e b ) : e vector<b>
+  // Copy will only happen on the first iteration in the map, to satisfy the unique precondition
```

Comment:
I don't understand this comment: With non-linear control, a copy could happen each iteration?

Reply:
Leftover from before I realized that I needed an ensure-unique after the call to `f`.